### PR TITLE
#4 filter recommendation  through category buttons

### PR DESCRIPTION
**Pull request by raniarusly**

### CODE REVIEW

File: src/main/webapp/scripts/map.js [hunk 1]
```diff
@@ -96,15 +96,20 @@ function initMap() {
   map.addListener("click", e => {
     placeMarkerAndPanTo(e.latLng, map);
   });
-  
+
   /* Add Hard-Coded Markers */
-  addMarker(map);
+  if(category) {
```
Comment (lornaarmstrong) [inline src/main/webapp/scripts/map.js]: maybe add a comment here to explain the code a little more

**Review by lornaarmstrong — COMMENTED** [review]
LGTM! Just a note to help my understanding but how does the filtering work? It seems like the user clicks a button and then the category of recommendations are added to the map, and the map is reloaded, rather than the other markers just being hidden?

Also, is there any way for the user to unselect/click the button again to see all the markers on the map? This is just an idea for future implementation.

**Comment by raniarusly** [comment]
> LGTM! Just a note to help my understanding but how does the filtering work? It seems like the user clicks a button and then the category of recommendations are added to the map, and the map is reloaded, rather than the other markers just being hidden?

yup, I was following the logic that we agreed to on the previous team sync, again this can be discussed on the next team sync. 
Short explanation about the logic:
- **markers** are pre-fetched when the page is loaded
- initMap() does not fetch all the markers again, instead just re-initialize the map with filtered (and pre-fetched) **markers**
Hence this logic only requires one call to the server, which is when loading the page only, not when clicking the button.
@MostafaGabriel might want to add more explanation or fix mine in case I am mistaken, thank you! :D

> Also, is there any way for the user to unselect/click the button again to see all the markers on the map? This is just an idea for future implementation.

I had this idea in mind too, an option will be adding a category button which says "all" and we can just call initMap() with no parameter, just like when loading the page on the first time.

File: src/main/webapp/scripts/map.js [hunk 2]
```diff
@@ -97,15 +97,18 @@ function initMap(category) {
     placeMarkerAndPanTo(e.latLng, map);
   });
 
-  /* Add Hard-Coded Markers */
+  /* Filter out the recommendations by category if a category parameter is passed */
+  /* this is the case when a category button is clicked */
   if(category) {
     addMarker(map, recommendations.filter(recommendation => recommendation.category === category));
   }
+  /* Add all recommendations if there is no parameter given (when loading the page)*/
```
Comment (lornaarmstrong) [inline src/main/webapp/scripts/map.js]: nit: 
```suggestion
  /* Add all recommendations if there is no parameter given (when loading the page) */
```

**Review by lornaarmstrong — APPROVED** [review]
LGMT! Thanks for the clarification!

File: src/main/webapp/scripts/map.js [hunk 3]
```diff
@@ -96,15 +96,23 @@ function initMap() {
   map.addListener("click", e => {
     placeMarkerAndPanTo(e.latLng, map);
   });
-  
-  /* Add Hard-Coded Markers */
-  addMarker(map);
+
+  /* Filter out the recommendations by category if a category parameter is passed */
+  /* this is the case when a category button is clicked */
```
Comment (MostafaGabriel) [inline src/main/webapp/scripts/map.js:105]: nit: we usually use the `//` style for one line comments but it's really up to you :)

File: src/main/webapp/scripts/map.js [hunk 4]
```diff
@@ -96,15 +96,23 @@ function initMap() {
...
+  if(category) {
```
Comment (MostafaGabriel) [inline src/main/webapp/scripts/map.js]:
```suggestion
  if (category) {
```
[4.6.2 Horizontal whitespace](https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace)

File: src/main/webapp/scripts/map.js [hunk 5]
```diff
@@ -96,15 +96,23 @@ function initMap() {
...
+  else {
+    addMarker(map, recommendations);
+  }
 }
 
-function addMarker(map) {
-  /* Iterate through stored recommendations to get details */
-  for (i = 0; i < recommendations.length; i++) {
-    const recommendation = recommendations[i];
+/* Add Hard-Coded Markers to map */
```
Comment (MostafaGabriel) [inline src/main/webapp/scripts/map.js:115]: Please use [JSDoc](https://google.github.io/styleguide/jsguide.html#jsdoc) here

**Review by MostafaGabriel — APPROVED** [review]
LGTM! Thanks Rania :)

**Pull request merged**